### PR TITLE
Document `Unaligned`

### PR DESCRIPTION
**Pull request by jswrenn**

### CODE REVIEW

**Review by joshlf — CHANGES_REQUESTED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -3219,26 +3215,135 @@ pub unsafe trait IntoBytes {
...
 pub use IntoBytes as AsBytes;
 
-/// Types with no alignment requirement.
+/// Analyzes whether a type is [`Unaligned`].
 ///
-/// WARNING: Do not implement this trait yourself! Instead, use
-/// `#[derive(Unaligned)]` (requires the `derive` Cargo feature).
+/// This derive analyzes, at compile time, whether the annotated type satisfies
+/// the [safety conditions] of `Unaligned` and implements `Unaligned` if it is sound
+/// to do so. This derive can be applied to structs, enums, and unions; e.g.:
```
[the remark this answers is not in the document]
Comment (jswrenn) [reply src/lib.rs]: Done

File: src/lib.rs [hunk 2]
```diff
@@ -3219,26 +3215,135 @@ pub unsafe trait IntoBytes {
...
+///
+/// If a type has the following properties, then this derive can implement
+/// `Unaligned` for that type:
+///
+/// - If the type is a struct or union:
+///   - If `repr(align(N))` is provided, `N` must equal 1.
+///   - If the type is `repr(C)` or `repr(transparent)`, all fields must be
+///     [`Unaligned`].
+///   - If the type is not `repr(C)` or `repr(transparent)`, it must be
+///     `repr(packed)`.
```
[the remark this answers is not in the document]
Comment (jswrenn) [reply src/lib.rs]: Done

File: src/lib.rs [hunk 3]
```diff
@@ -3219,26 +3215,135 @@ pub unsafe trait IntoBytes {
...
+/// - If the type is an enum:
+///   - If `repr(align(N))` is provided, `N` must equal 1.
+///   - It must be a C-like enum (meaning that all variants have no fields).
```
[the remark this answers is not in the document]
Comment (joshlf) [reply src/lib.rs:3273]: Ack.

File: src/lib.rs [hunk 4]
```diff
@@ -3219,26 +3215,136 @@ pub unsafe trait IntoBytes {
...
+///
+/// If a type has the following properties, then this derive can implement
+/// `Unaligned` for that type:
+///
+/// - If the type is a struct or union:
+///   - If `repr(align(N))` is provided, `N` must equal 1.
+///   - If the type is `repr(C)` or `repr(transparent)`, all fields must be
+///     [`Unaligned`].
+///   - If the type is not `repr(C)` or `repr(transparent)`, it must be
+///     `repr(packed)` or ``repr(packed(1))`.
```
[the remark this answers is not in the document]
Comment (jswrenn) [reply src/lib.rs]: Whoops; done.

**Review by joshlf — APPROVED** [review]

**Pull request merged**